### PR TITLE
feat!: Consume SDK interface changes

### PR DESCRIPTION
**Pull request by weichou1229**
BREAKING CHANGE: Consume SDK interface changes by adding Discover and ValidateDevice func on driver

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) not impact
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?) not impact
  <link to docs PR>

## Testing Instructions

Run core service and device service to verify the func works.

### CODE REVIEW

File: internal/driver/driver.go [hunk 1]
```diff
@@ -412,8 +415,7 @@ func (d *Driver) Discover() {
 	d.configMu.RUnlock()
 
 	if !discoveryMode.IsValid() {
-		d.lc.Errorf("DiscoveryMode is set to an invalid value: %s. Refusing to do discovery.", discoveryMode)
-		return
+		return fmt.Errorf("discoveryMode is set to an invalid value: %s. Refusing to do discovery", discoveryMode)
```
Comment (ajcasagrande) [inline internal/driver/driver.go]: `discoveryMode` is based on `d.config.AppCustom.DiscoveryMode`, which I believe is still uppercase in the configuration, so I would leave it uppercase in the error message: `DiscoveryMode`

File: internal/driver/driver.go [hunk 2]
```diff
@@ -208,7 +208,10 @@ func (d *Driver) debouncedDiscover() {
 			d.debounceTimer = nil
 			d.debounceMu.Unlock()
 
-			d.Discover()
+			err := d.Discover()
+			if err != nil {
+				d.lc.Errorf("failed to discover device, %v", err)
```
Comment (ajcasagrande) [inline internal/driver/driver.go]: I think something more like: `failed to run device discovery` sounds better.

**Review by ajcasagrande — CHANGES_REQUESTED** [review]

**Review by presatish — DISMISSED** [review]

File: internal/driver/driver.go (hunk 1 above)
Comment (weichou1229) [reply internal/driver/driver.go]: Updated

File: internal/driver/driver.go (hunk 2 above)
Comment (weichou1229) [reply internal/driver/driver.go]: Updated

**Review by ajcasagrande — APPROVED** [review]

**Pull request merged**